Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -220,7 +220,7 @@ static ALWAYS_INLINE bool isPrivateFieldName(UniquedStringImpl* uid)
 }
 
 template <typename LexerType>
-Expected<typename Parser<LexerType>::ParseInnerResult, String> Parser<LexerType>::parseInner(const Identifier& calleeName, ParsingContext parsingContext, std::optional<int> functionConstructorParametersEndPosition, const FixedVector<JSTextPosition>* classFieldLocations, const PrivateNameEnvironment* parentScopePrivateNames)
+Expected<typename Parser<LexerType>::ParseInnerResult, String> Parser<LexerType>::parseInner(const Identifier& calleeName, ParsingContext parsingContext, ImplementationVisibility implementationVisibility, std::optional<int> functionConstructorParametersEndPosition, const FixedVector<JSTextPosition>* classFieldLocations, const PrivateNameEnvironment* parentScopePrivateNames)
```

Comment:
We do not need to pass this parameter to every parse functions. We should store it to the FunctionScope when parsing a function. And when creating inner functions, let's get it from `currentFunctionScope()`. And from this outer function's visibility, we can get the visibility for inner functions. This scope chain may not include all scopes up to the top-level. But at least, it includes chains up to the most closest function, and that's enough.
See `ConstructorKind` for example.

Reply:
For functions in Program / ModuleProgram / Eval etc., probably a bit different thing is required, but basically, putting it into scope would be the right way.